tc/core/polyhedral/cuda/memory_promotion_heuristic.cc

```diff
@@ -323,6 +323,24 @@ bool promotionImprovesCoalescing(
   return false;
 }
 
+/*
+ * Returns the union of all mapping filters to "MappingType" in "scop".
+ */
+template <typename MappingType>
+isl::union_set collectMappingsTo(const Scop& scop) {
+  auto root = scop.scheduleRoot();
+  auto domain = scop.domain();
+  auto mappingFilters = detail::ScheduleTree::collect(
+      root, detail::ScheduleTreeType::MappingFilter);
+  mappingFilters = functional::Filter(isMappingTo<MappingType>, mappingFilters);
+  auto mapping = isl::union_set::empty(domain.get_space());
+  for (auto mf : mappingFilters) {
+    auto filterNode = mf->elemAs<detail::ScheduleTreeElemMappingFilter>();
+    mapping = mapping.unite(filterNode->filter_);
+  }
+  return domain.intersect(mapping);
```

Comment:
Why do you need to intersect with the domain?

Reply:
To have the domain be consistent with the domain of the function returned by `extractDomainToIds`

Reply:
On Fri, Jun 15, 2018 at 01:33:06AM -0700, ftynse wrote:
> To have the domain be consistent with the domain of the function returned by `extractDom
ainToIds`

Mention that in the commit message and mention
the intersection in the comment.